src/global_mapper_ros/global_mapper_ros.cc

```diff
@@ -103,17 +201,25 @@ void GlobalMapperRos::PointCloudCallback(const pcl::PointCloud<pcl::PointXYZ>::C
   pcl::PointCloud<pcl::PointXYZ>::Ptr cloud_trans_ptr = cloud_trans.makeShared();
   pcl_ros::transformPointCloud(*cloud_ptr, *cloud_trans_ptr, transform);
 
+  // set origin of pointcloud
+  // NOTE(jakeware): feel like there must be a better way to do this.
+  cloud_trans_ptr->sensor_origin_[0] = transform_stamped.transform.translation.x;
+  cloud_trans_ptr->sensor_origin_[1] = transform_stamped.transform.translation.y;
+  cloud_trans_ptr->sensor_origin_[2] = transform_stamped.transform.translation.z;
+
   // push to mapper
-  global_mapper_.PushPointCloud(cloud_trans_ptr);
+  global_mapper_ptr_->PushPointCloud(cloud_trans_ptr);
 }
 
 void GlobalMapperRos::Run() {
-  GetParams();
+  GlobalMapperParams params;
+  GetParams(params);
```

Comment:
I'm trying to do a better job of writing functional code and keeping the constructors empty as a stylistic thing.  I'm doing better on the second part of that than the first thus far.